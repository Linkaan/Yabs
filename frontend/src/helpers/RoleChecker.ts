import UsersModule, { User } from '../store/modules/UsersModule';

export default class RoleChecker {

  private static checkPermission(user: User, required: number): boolean {
    return user && ((user.role & required) == required || (user.role > required));
  }
  
  static isPrincipal(): boolean {
    return this.checkPermission(UsersModule.currentUser, 32);
  }

  static isAdmin(): boolean {
    return this.checkPermission(UsersModule.currentUser, 16);
  }

  static isCaretaker(): boolean {
    return this.checkPermission(UsersModule.currentUser, 8);
  }

  static isTeacher(): boolean {
    return this.checkPermission(UsersModule.currentUser, 4);
  }

  static isStudentHealth(): boolean {
    return this.checkPermission(UsersModule.currentUser, 2);
  }

  static isStudent(): boolean {
    return UsersModule.currentUser && (UsersModule.currentUser.role & 1) == 1;
  }
}
