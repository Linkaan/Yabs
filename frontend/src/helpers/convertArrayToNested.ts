export default function convertList(payload: any[], typeOfId: string): object { //eslint-disable-line
  return payload.reduce((acc, item) => {
    acc[item[typeOfId]] = item;
    return acc;
  }, {});
}
