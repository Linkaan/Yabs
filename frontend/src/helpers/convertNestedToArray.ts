export default function convertList(payload: any, typeOfId: string): any[] { //eslint-disable-line
  const res = Object.keys(payload).map((id) => payload[id]);
  return res;
}
