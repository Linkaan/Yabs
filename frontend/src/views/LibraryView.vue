<template>
  <v-container> 
    <ReturnLoanComponent />
    <h2>Lägg till bock att låna</h2>
    <LibraryLoanComponent />
    <h2>Dina lån</h2>
    <LoanListComponent />
    <h2>Sök efter böcker</h2> 
    <TitleListComponent />
  </v-container>
</template>

<script lang='ts'>
import { defineComponent, SetupContext } from '@vue/composition-api';
import LoanListComponent from '@/components/LoanListComponent.vue';
import LibraryLoanComponent from '@/components/LibraryLoanComponent.vue';
import TitleListComponent from '@/components/TitleListComponent.vue';
import ReturnLoanComponent from '@/components/ReturnLoanComponent.vue';
import LoansModule from '../store/modules/LoansModule';
import TitlesModule from '../store/modules/TitlesModule';
import BooksModule from '../store/modules/BooksModule';
import { VuexModule } from 'vuex-module-decorators';

export default defineComponent({
  name: 'LibraryView',
  components: {
    LoanListComponent,
    LibraryLoanComponent,
    TitleListComponent,
    ReturnLoanComponent,
  },
  setup(_ : object, { root } : SetupContext) : object {
    const loansModule: VuexModule = LoansModule;
    const titlesModule: VuexModule = TitlesModule;
    const booksModule: VuexModule = BooksModule;

    function created() : void {
      LoansModule.fetchAll(); 
      TitlesModule.fetchAll();
      BooksModule.fetchAll();
    }

    created();

    return { loansModule, titlesModule, booksModule, created };
  }
});
</script>

<style>

</style>