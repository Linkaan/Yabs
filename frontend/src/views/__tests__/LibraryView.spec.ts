import { shallowFactory } from '@/helpers/testFactoryHelpers';
import { Wrapper } from '@vue/test-utils';
import LibraryView from '../LibraryView.vue';
import Vue from 'vue';

describe('LibraryView.vue', () => {
  it('is not empty', () => {
    const wrapper: Wrapper<Vue> = shallowFactory(LibraryView);
    expect(wrapper.html()).not.toBe('');
  });
  it('matches the snapshot', async () => {
    const wrapper: Wrapper<Vue> = shallowFactory(LibraryView);
    expect(wrapper.html()).toMatchSnapshot();
  });
});
