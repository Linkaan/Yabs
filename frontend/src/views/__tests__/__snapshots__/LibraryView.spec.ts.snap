// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LibraryView.vue matches the snapshot 1`] = `
<v-container-stub tag="div">
  <returnloancomponent-stub></returnloancomponent-stub>
  <h2>Add loan</h2>
  <libraryloancomponent-stub></libraryloancomponent-stub>
  <h2>Your loans</h2>
  <loanlistcomponent-stub></loanlistcomponent-stub>
  <h2>Search for a book</h2>
  <titlelistcomponent-stub></titlelistcomponent-stub>
</v-container-stub>
`;
